libbeat/cmd/instance/metrics/metrics.go

```diff
@@ -118,129 +119,65 @@ func reportMemStats(m monitoring.Mode, V monitoring.Visitor) {
 }
 
 func getRSSSize() (uint64, error) {
-	state, err := getBeatProcessState()
+	state, err := beatProcessStats.GetSelf()
 	if err != nil {
 		return 0, err
 	}
 
-	iRss, err := state.GetValue("memory.rss.bytes")
-	if err != nil {
-		return 0, fmt.Errorf("error getting Resident Set Size: %v", err)
-	}
-
-	rss, ok := iRss.(uint64)
-	if !ok {
-		return 0, fmt.Errorf("error converting Resident Set Size to uint64: %v", iRss)
-	}
-	return rss, nil
+	return state.Memory.Rss.Bytes.ValueOr(0), nil
 }
 
-func getBeatProcessState() (common.MapStr, error) {
-	pid, err := process.GetSelfPid()
+// GetOwnResourceUsageTimeInMillis return the user and system CPU usage time in milliseconds
+func getOwnResourceUsageTimeInMillis() (int64, int64, error) {
```

Comment:
Maybe create a new type for the return value:
```golang
type RUsageMS struct {
    User opt.Int
    System opt.Int
}
```

Reply:
So, I just realized that the ticks value we're reporting is _also_ milliseconds, so I've just duplicated the `ticks` value here across to `ms`, since I'm not sure if I can make breaking changes to those metrics namespaces.